system/security/zcrypt/ioapi.c

```diff
@@ -140,9 +139,7 @@ long ZCALLBACK fseek_file_func (opaque, stream, offset, origin)
         break;
     default: return -1;
     }
-    ret = 0;
-    fseek((FILE *)stream, offset, fseek_origin);
-    return ret;
+    return (long)fseek((FILE *)stream, offset, fseek_origin);
```

Comment:
This will remove the warning and change the behaviour... have all calls to this function been checked (that they are ok with a non-zero result should they seek anywhere other than the start of the file).

Reply:
I'm not convinced that changing the functionality of this function to silence a warning is a good idea. This is third-party code - unless we believe that the issue coverity spotted is going to cause us any problem, normal practice would be to mark it as "ignore" or "false positive" as appropriate in the coverity system.